teps/0093-tkn-cli-extension-for-signverify.md

```diff
@@ -0,0 +1,347 @@
+6. Provide an option to convert tkn definitions to intoto-layout first and then sign it.
+
+#### `tkn verify`
+```bash
+% tkn verify -h
+USAGE
+  tkn verify -key <key path> [-r] <pipeline dir> [i] <oci registry path> [t] <image tag>
+
+Verify all tekton pipeline resources
+EXAMPLES
+  # verify all pipeline resources
+  tkn verify -k ./public.pub -d ./sample-pipeline-dir -i us.icr.io/tap8stry -t dev1
+  # verify resources for a give pipeline
+  tkn verify -k ./public.pub -d ./sample-pipeline-dir -i us.icr.io/tap8stry -t dev1 -p pr-pipeline
+
+
+FLAGS
+  -d ...    pipeline directory
+  -i ...    oci image registry path
+  -key ...  path to the private key file, KMS URI or Kubernetes Secret
+  -r=false  scan all pipeline resources recusively
+  -t ...    oci image path to use
+```
+
+This command would perform following actions (these are counter actions to `sign` command):
+
+1. recursively traverse all resources and parse their signatures from the metadata
+2. fetch their signatures from the specified backend
+3. verify signatures against given artifacts
+4. Report pass/fail status for every resource 
+
+
+### Notes/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate. Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the WGs or subproject.
+-->
+
+### User Experience (optional)
+
+<!--
+Consideration about the user experience. Depending on the area of change,
+users may be task and pipeline editors, they may trigger task and pipeline
+runs or they may be responsible for monitoring the execution of runs,
+via CLI, dashboard or a monitoring system.
+
+Consider including folks that also work on CLI and dashboard.
+-->
+
+### Performance (optional)
+
+<!--
+Consideration about performance.
+What impact does this change have on the start-up time and execution time
+of task and pipeline runs? What impact does it have on the resource footprint
+of Tekton controllers as well as task and pipeline runs?
+
+Consider which use cases are impacted by this change and what are their
+performance requirements.
+-->
+
+## Design Details
+
+Once task/pipeline YAML file is signed, it will be updated with following annotations
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  annotations:
+    cosign.sigstore.dev/imageRef: icr.io/gitsecure/git-clone-repo:demo1
+  name: git-clone-repo
+spec:
+```
+
+Annotations would indicate the backend that is used for storing the signature. For signing step images, we can follow 2 options:
```

Comment:
+1 I like the idea of having a configurable backend. One option could be store signatures along with the original resources. If the resource is stored in an OCI registry they could go there, but if the resource is stored in git or elsewhere the signature could live next to it.